src/api/java/com/minecolonies/api/util/MessageUtils.java

```diff
@@ -0,0 +1,153 @@
+    }
+
+    /**
+     * Send a message to the given player.
+     *
+     * @param player    the player.
+     * @param colony    the colony which sent the message.
+     * @param component the message component.
+     */
+    public static void sendPlayerColonyMessage(PlayerEntity player, IColony colony, ITextComponent component)
+    {
+        sendPlayerColonyMessageWithColor(player, colony, TextFormatting.WHITE, component);
+    }
+
+    /**
+     * Send a message to the given player.
+     *
+     * @param player the player.
+     * @param colony the colony which sent the message.
+     * @param key    the translation component key.
+     * @param args   the additional args to send in the translation component.
+     */
+    public static void sendPlayerColonyMessageWithColor(PlayerEntity player, IColony colony, TextFormatting color, String key, Object... args)
+    {
+        sendPlayerColonyMessageWithColor(player, colony, color, new TranslationTextComponent(key, args));
+    }
+
+    /**
+     * Send a message to the given player.
+     *
+     * @param player    the player.
+     * @param colony    the colony which sent the message.
+     * @param component the message component.
+     */
+    public static void sendPlayerColonyMessageWithColor(PlayerEntity player, IColony colony, TextFormatting color, ITextComponent component)
+    {
+        ITextComponent fullComponent = component;
+
+        if (colony != null && !colony.isCoordInColony(player.level, player.blockPosition()))
+        {
+            fullComponent = new StringTextComponent("[" + colony.getName() + "] ")
+                              .withStyle(TextFormatting.WHITE)
+                              .append(component);
+        }
+
+        sendMessageInternal(player, fullComponent, Collections.singletonList(color));
+    }
+
+    /**
+     * Internal message sending handler.
+     *
+     * @param player     the player to send to.
+     * @param component  the text component to send.
+     * @param formatters a collection of formatter to apply to the root component.
+     */
+    private static void sendMessageInternal(PlayerEntity player, ITextComponent component, Collection<TextFormatting> formatters)
+    {
+        if (player == null)
+        {
+            Log.getLogger().log(Level.ERROR, "Attempting to send player message without passing a player, please report this to the developers!");
+            return;
+        }
+        if (component == null)
+        {
+            Log.getLogger().log(Level.ERROR, "Attempting to send player message without passing a message component, please report this to the developers!");
+            return;
+        }
+
+        if (component instanceof IFormattableTextComponent)
+        {
+            for (TextFormatting formatter : formatters)
+            {
+                ((IFormattableTextComponent) component).withStyle(formatter);
```

Comment:
Meh, if I use the method with component I expect it to not change, so rather than applying it here I would make methods
sendXYmessage(player, key, args...) - no style modify
sendXYmessage(player, key, style, args...) - rather than color you should use style directly (if textformatting does not implement style then unfortunately another method overload)
sendXYmessage(player, component) - no style modify
sendXYmessage(player, style, component) - makes little sense since if you already have component then you can style it urself

also not sure why this **private** method use list when you do `Collections.singletonList` everywhere